File: cmps12b/Lab3/fileReverse.c
/*
     Name: Johnson Le
     cruzID: jonle
     id number: 1512346
     File Name: fileReverse.c
     Description: takes tokens and reverses it. ie abc-> cba.
*/

#include<stdio.h>
#include<stdlib.h>
#include<string.h>

// char test[]="abcdefghijklmnopqrstuvwxyz";
char string[256];
void stringReverse(char* s);

int main(int argc, char* argv[]){
     // printf("normal: %s", test);
     // stringReverse(test);
     //declaring the files.
     FILE* inputFile;
     FILE* outputFile;
     //following code is provided code from assignment
     /* check command line for correct number of arguments */
     if( argc != 3 ){
        printf("Usage: %s <input file> <output file>\n", argv[0]);
        exit(EXIT_FAILURE);
     }
     /* open input file for reading */
     inputFile = fopen(argv[1], "r");
     if( inputFile==NULL ){
        printf("Unable to read from file %s\n", argv[1]);
        exit(EXIT_FAILURE);
     }

     /* open output file for writing */
     outputFile = fopen(argv[2], "w");
     if( outputFile==NULL ){
        printf("Unable to write to file %s\n", argv[2]);
        exit(EXIT_FAILURE);
     }
     //following is changed a bit to reverse.
     while( fscanf(inputFile, " %s", string) != EOF ){//eof = end of file
          stringReverse(string);
          fprintf(outputFile, "%s\n", string);
          // printf("%s\n", string);
     }


     return(EXIT_SUCCESS);
}

void stringReverse(char* s){
     //because there is no return type,
     //have to work with pointers.

     char *first = s;                   //points to the first address in s
     char *last = first+strlen(s)-1;    //points to the last
     char temp;                         //will recycle

     while(last>first){                 //basically swapping
          temp = *first;                //temp=first char
          *first = *last;               //now first char replaced by last
          *last = temp;                 //last char is temp=first char
          first=first+1;
          last = last-1;
     }
     // printf("\nreverse: %s", s);
}

File: cmps12b/Lab2/FileReverse.java
//----------------------------------------------------------------------
// Name: Johnson Le
// cruzID: jonle
// id number: 1512346
// File Name: FileReverse.java
// Description: takes tokens and reverses it. ie abc-> cba.
//----------------------------------------------------------------------

import java.io.*;
import java.util.Scanner;

class FileReverse{
     public static void main(String[] args) throws IOException{
          int lineNumber = 0;
          // check number of command line argument is at least 2
          if(args.length <2){
               System.out.println("Usage: java -jar FileReverse.jar <input file> <output file>");
               System.exit(1);
          }
          // open file
          Scanner in = new Scanner(new File(args[0]));
          PrintWriter out = new PrintWriter(new FileWriter(args[1]));

          while(in.hasNextLine()){
               lineNumber++;

               String line = in.nextLine().trim() + " "; //think white space? and adds 1? idk lol.
               String[] token = line.split("\\s+");    //splits based on whitespace. the \\s
                                                       //  \\s means 1 white. \\s+ means more

               int n = token.length; //length of token array
               for(int i=0; i<n; i++){
                    String reverse = stringReverse(token[i]);
                    token[i]= reverse;
                    out.println("  "+token[i]);
               }
          }
          in.close();
          out.close();
     }
     public static String stringReverse(String s){
          String backwards = "";
          for (int i=s.length()-1 ; i>=0 ; i--){
               backwards += s.charAt(i);
          }
          return backwards;
     }
}
